src/storage/irc_table_set.cpp

```diff
@@ -29,14 +29,21 @@ namespace duckdb {
 ICTableSet::ICTableSet(IRCSchemaEntry &schema) : schema(schema), catalog(schema.ParentCatalog()) {
 }
 
-bool ICTableSet::FillEntry(ClientContext &context, IcebergTableInformation &table) {
+bool ICTableSet::FillEntry(ClientContext &context, IcebergTableInformation &table, bool might_throw) {
 	if (!table.schema_versions.empty()) {
 		return true;
 	}
 
 	auto &ic_catalog = catalog.Cast<IRCatalog>();
-
-	auto get_table_result = IRCAPI::GetTable(context, ic_catalog, schema, table.name);
+	APIResult<rest_api_objects::LoadTableResult> get_table_result;
+	try {
+		get_table_result = IRCAPI::GetTable(context, ic_catalog, schema, table.name);
+	} catch (...) {
+		if (!might_throw) {
```

Comment:
I worry if this is ever caught? I don't believe `IRCAPI::GetTable` will throw, I think all the information is in the `APIResult`